### ISSUE TITLE
Addon-knobs label default value persisting

### ISSUE THREAD

**Issue by craigtaub** [storybookjs/storybook#7150]
Hi 
We noticed since the 5.0.11 to 5.1.8 upgrade the `text` label default value persists across files/stories. We use `Heading` quite often and now sometimes the default set in 1 Story appears in a different Story (even if it has a different default set). The result is that our Chromatic flags a change in the UI. Happens locally sometimes too. Produces a race condition regarding which file loads first (and sets the default).

I could not find much in the documentation about how this should be handled but I assumed it should let you re-use label names cross file/story.

**To Reproduce**
Removed most superfluous code in attempt to simplify.

- Install `addon-knobs`
- Import `text`
- Create two variables under the same label name in 1 file
- Create a third variable under the same label name in another file
- The default value set from 1 file persists to the other

**Expected behavior**
I would have expected FileB.stories.js `three` value to be `third heading`

**Code snippets**

FileA.stories.js
``` javascript
import React from "react";
import { text } from "@storybook/addon-knobs";

const one = text("Input", "first heading");
const two = text("Input", "second heading");

console.log(one); // "first heading"
console.log(two); // "first heading". Would kind-of expect this so use a different label name.
```

FileB.stories.js
``` javascript
import React from "react";
import { text } from "@storybook/addon-knobs";

const three = text("Input", "third heading");

console.log(three); // "first heading". Would not expect this.
```

**System:**
 - OS: MacOS (also happening on Chromatic)
 - Device: Macbook Pro 2018
 - Browser: Chrome
 - Framework: React 
 - Addons: knobs, text
 - Version: below

```
"@storybook/addon-a11y": "5.1.8",
"@storybook/addon-actions": "5.1.8",
"@storybook/addon-knobs": "5.1.8",
"@storybook/addon-links": "5.1.8",
"@storybook/addon-viewport": "5.1.8",
"@storybook/react": "5.1.8",
```

**Comment by craigtaub** [comment]
@Armanio @leoyli any chance someone could have a quick look at my question please? Would really appreciate it.

**Comment by allpwrfulroot** [comment]
I'm also seeing this, updating button text for example is unresponsive.

**Comment by craigtaub** [comment]
Anyone able to help here? Been waiting 2 months.

**Comment by shilman** [comment]
Hi gang, We’ve just released addon-controls in 6.0-beta!

Controls are portable, auto-generated knobs that are intended to replace addon-knobs long term.

- [Controls README](https://github.com/storybookjs/storybook/tree/next/addons/controls)
- [Migration instructions](https://github.com/storybookjs/storybook/blob/next/addons/controls/README.md#how-do-i-migrate-from-addon-knobs)

Please upgrade and try them out today. Thanks for your help and support getting this stable for release!

**Comment by craigtaub** [comment]
Ok thanks
